controllers/front/ProductController.php

```diff
@@ -806,14 +806,16 @@ public function getBreadcrumbLinks()
     {
         $breadcrumb = parent::getBreadcrumbLinks();
 
-        $categoryDefault = new Category($this->product->id_category_default);
+        $categoryDefault = new Category($this->product->id_category_default, $this->context->language->id);
```

Comment:
The best way would be to put this `(int)` cast inside the constructor called, not in this line. This will make all your calls "secured".